e2e-tests/gef/cypress/e2e/application-details.spec.js

```diff
@@ -311,4 +312,25 @@ context('Application Details Page', () => {
       });
     });
   });
+
+  describe('Restrict which users can access page', () => {
+    const authorisedRoles = [
+      {
+        roleName: 'admins',
+        userWithRole: ADMIN,
+      },
+      {
+        roleName: 'Read only users with access to all banks',
+        userWithRole: READ_ONLY_ALL_BANKS,
+      },
+    ];
+
+    authorisedRoles.forEach(({ roleName, userWithRole }) => {
+      it(`allows ${roleName} to access the gef deals page`, () => {
+        cy.login(userWithRole);
+        cy.visit(relative(`/gef/application-details/${dealWithEmptyExporter._id}`));
+        cy.url().should('eq', relative(`/gef/application-details/${dealWithEmptyExporter._id}`));
+      });
+    });
```

Comment:
Love it -- This is also built into jest btw so you could have:

```suggestion
      it.each(authorisedRoles)(`allows ${roleName} to access the gef deals page`, (userWithRole) => {
        cy.login(userWithRole);
        cy.visit(relative(`/gef/application-details/${dealWithEmptyExporter._id}`));
        cy.url().should('eq', relative(`/gef/application-details/${dealWithEmptyExporter._id}`));
      });
```

https://jestjs.io/docs/api#testeachtablename-fn-timeout